src/template.c

```diff
@@ -5956,22 +5956,20 @@ void TemplateInstance::semantic(Scope *sc, Expressions *fargs)
         this->tnext = inst->tnext;
         inst->tnext = this;
 
-        // If the first instantiation was in speculative context, but this is not:
-        if (tinst && !inst->tinst && !inst->minst)
+        if (minst && minst->isRoot() && !(inst->minst && inst->minst->isRoot()))
```

Comment:
> A simple bool isSpeculative() { return minst == NULL; } would help.

Unfortunately we already have `Dsymbol::isSpeculative()` and its meaning is not same with the condition tested in here. I'd really want to determine good name, but it's not discovered yet.